controllers/credential_controller.go

```diff
@@ -207,3 +207,14 @@ func EditCredential(ctx *gin.Context) {
 
 	SendResponse(ctx, 200, nil, "", nil)
 }
+
+func GetSearchData(ctx *gin.Context) {
+
+	caller, err := utils.FetchUserIDFromCtx(ctx)
+	credentials, err := service.GetSearchData(ctx, caller)
+	if err != nil {
+		SendResponse(ctx, 200, nil, "Failed to fetch credential", errors.New("failed to fetch credential"))
```

Comment:
```suggestion
		SendResponse(ctx, 500, nil, "",  err)
```
You can send the original error back for debugging. It is not necessary. Upto you. I am currently sending the original errors and plan to remove them once we are past alpha